wpilibc/src/main/native/cpp/shuffleboard/ShuffleboardContainer.cpp

```diff
@@ -0,0 +1,167 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "frc/shuffleboard/ShuffleboardContainer.h"
+
+#include <wpi/raw_ostream.h>
+
+#include "frc/shuffleboard/ComplexWidget.h"
+#include "frc/shuffleboard/ShuffleboardComponent.h"
+#include "frc/shuffleboard/ShuffleboardLayout.h"
+#include "frc/shuffleboard/SimpleWidget.h"
+
+using namespace frc;
+
+ShuffleboardContainer::ShuffleboardContainer(wpi::StringRef title)
+    : ShuffleboardValue(title) {}
+
+ShuffleboardContainer::ShuffleboardContainer(ShuffleboardContainer&& rhs)
+    : ShuffleboardValue(rhs) {
+  m_usedTitles = std::move(rhs.m_usedTitles);
+  m_components = std::move(rhs.m_components);
+  m_layouts = std::move(rhs.m_layouts);
+}
+
+std::vector<std::unique_ptr<ShuffleboardComponentBase>>&
+ShuffleboardContainer::GetComponents() {
```

Comment:
Does this have to be non-const?  Isn't it a little dangerous to return direct access to the underlying vector of unique_ptrs?

Reply:
Yes and yes. It gets mutated somewhere and caused a compiler error when I tried to const qualify it a few days ago. Seems like a code smell, but I was tired of fighting the compiler for a week straight. :man_shrugging: